stm32mp1.mk

```diff
@@ -0,0 +1,212 @@
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER   := 32
+override COMPILE_NS_KERNEL := 32
+override COMPILE_S_USER    := 32
+override COMPILE_S_KERNEL  := 32
+
+PLATFORM ?= stm32mp1-157C_DK2
+OPTEE_OS_PLATFORM := $(PLATFORM)
+
+include common.mk
+
+ifeq ($(PLATFORM),stm32mp1-157A_DK1)
+BREXT_FLAVOR		= STM32MP157A-DK1
+STM32MP1_DTS_BASENAME	= stm32mp157a-dk1
+else ifeq ($(PLATFORM),stm32mp1-157C_DK2)
+BREXT_FLAVOR		= STM32MP157C-DK2
+STM32MP1_DTS_BASENAME	= stm32mp157c-dk2
+else ifeq ($(PLATFORM),stm32mp1-157C_EV1)
+BREXT_FLAVOR		= STM32MP157C-EV1
+STM32MP1_DTS_BASENAME	= stm32mp157c-ev1
+else
+$(error Unknown PLATFORM $(PLATFORM))
+endif
+
+################################################################################
+# Binary images names
+################################################################################
+
+ARM_TF_BIN		:= tf-a-$(STM32MP1_DTS_BASENAME).stm32
+OPTEE_HEADER_BIN	:= tee-header_v2.stm32
+OPTEE_PAGER_BIN		:= tee-pager_v2.stm32
+OPTEE_PAGEABLE_BIN  	:= tee-pageable_v2.stm32
+U_BOOT_BIN		:= u-boot.stm32
+LINUX_KERNEL_BIN 	:= uImage
+LINUX_DTB_BIN		:= $(STM32MP1_DTS_BASENAME).dtb
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+BINARIES_PATH		?= $(ROOT)/out/bin
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+U_BOOT_PATH		?= $(ROOT)/u-boot
+
+define install_in_binaries
+	echo "  INSTALL $(shell basename $1) to $(BINARIES_PATH)" && \
+	mkdir -p $(BINARIES_PATH) && \
+	ln -sf $1 $(BINARIES_PATH)
+endef
+
+################################################################################
+# Main targets
+################################################################################
+all: arm-tf optee-os u-boot linux buildroot
+	@$(call install_in_binaries,$(ROOT)/out-br/images/sdcard.img)
+	@echo Build for platform $(PLATFORM) completed
+
+clean: arm-tf-clean optee-os-clean u-boot-clean linux-clean buildroot-clean \
+       check-clean
+
+include toolchain.mk
+
+################################################################################
+# OP-TEE OS
+################################################################################
+optee-os: optee-os-common
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_HEADER_BIN))
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_PAGER_BIN))
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_PAGEABLE_BIN))
+
+optee-os-clean: optee-os-clean-common
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+TF_A_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
```

Comment:
I've suggested "tfa" etc above, here you have "tf_a". Both will do, pick one of them and use the same everywhere.

Reply:
Agree, i'll make it more consistent. thanks.